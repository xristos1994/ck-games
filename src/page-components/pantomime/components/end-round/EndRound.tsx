import React, { FC, ReactElement } from "react";
import { connect } from "react-redux";
import { createStructuredSelector } from "reselect";
import { Button } from "@components";
import { classnames, compose } from "@utils/component-utils";
import { withTranslation, ITranslate } from "@models/i18n/hoc";
import { goToNextRound, setIfMovieFound } from "@models/pantomime/actions";
import { teams } from "@models/pantomime/props";
import { IState } from "@models/interfaces";
import { ScoreBoard } from "../score-board";
const styles = require("./styles.module.css");

interface IProps {
  teams: {
    id: number;
    score: number;
    name: string;
    playsNow: boolean | null;
    movieFound: boolean;
  }[];
  goToNextRound: () => void;
  setIfMovieFound: (boolean) => void;
  t: ITranslate;
}

const _EndRound: FC<IProps> = ({
  teams,
  goToNextRound,
  setIfMovieFound,
  t,
}): ReactElement => {
  const teamNow = teams.find(team => team.playsNow);

  return (
    <div className={styles.endRoundContainer}>
      <div className={classnames(styles.title)}>
        {t("Team, your turn completed", [teamNow.name])}
      </div>
      <ScoreBoard />
      <div className={styles.movieFoundButtons}>
        <Button
          key="movieFound"
          other={{ disabled: teamNow.movieFound }}
          className={classnames({
            [styles.movieActionSelected]: teamNow.movieFound,
            [styles.movieActionNotSelected]: !teamNow.movieFound,
          })}
          onClick={() => setIfMovieFound(true)}
        >
          {t("Movie Found")}
        </Button>
        <Button
          key="movieNotFound"
          other={{ disabled: !teamNow.movieFound }}
          className={classnames({
            [styles.movieActionSelected]: !teamNow.movieFound,
            [styles.movieActionNotSelected]: teamNow.movieFound,
          })}
          onClick={() => setIfMovieFound(false)}
        >
          {t("Movie Not Found")}
        </Button>
      </div>
      <Button
        onClick={() => goToNextRound()}
        className={classnames(styles.proceedButton)}
      >
        {t("CONTINUE")}
      </Button>
    </div>
  );
};

const EndRound = compose(
  connect(
    createStructuredSelector<
      IState,
      {
        teams: IProps["teams"];
      },
      {
        teams: IProps["teams"];
      }
    >({
      teams,
    }),
    { goToNextRound, setIfMovieFound }
  ),
  withTranslation
)(_EndRound);

export { EndRound };
