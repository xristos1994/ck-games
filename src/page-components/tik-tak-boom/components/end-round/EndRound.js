import React from "react";
import { connect } from "react-redux";
import { createStructuredSelector } from "reselect";
import { Button } from "@components";
import { classnames } from "@utils/component-utils";
import * as styles from "./styles.module.css";
import { goToNextRound, setWhoLost } from "@models/tik-tak-boom/actions";
import { players } from "@models/tik-tak-boom/props";

const _EndRound = ({ players, goToNextRound, setWhoLost }) => {
  const playerNow = players.find(player => player.playsNow);

  return (
    <div className={styles.endRoundContainer}>
      <div
        className={classnames(styles.whoLost, "extraLargeText", "main-color")}
      >
        Bomb boomed on {playerNow.name}'
        {playerNow.name.slice(-1).toLowerCase() === "s" ? "" : "s"} hands
      </div>
      <div className={styles.playersContainer}>
        {players.map(player => (
          <Button
            key={player.id}
            disabled={!player.isActive}
            className={classnames(styles.player, "largeText", {
              "secondary-light": playerNow.id === player.id,
              "secondary-dark": playerNow.id !== player.id,
            })}
            onClick={() => player.isActive && setWhoLost(player.id)}
          >
            {player.name}
          </Button>
        ))}
      </div>
      <Button
        onClick={() => goToNextRound()}
        className={classnames(
          styles.proceedButton,
          "primary",
          "extraLargeText"
        )}
      >
        PROCEED
      </Button>
    </div>
  );
};

const EndRound = connect(
  createStructuredSelector({
    players,
  }),
  { goToNextRound, setWhoLost }
)(_EndRound);

export { EndRound };
