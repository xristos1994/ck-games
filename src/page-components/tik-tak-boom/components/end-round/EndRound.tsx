import React, { FC, ReactElement } from "react";
import { connect } from "react-redux";
import { createStructuredSelector } from "reselect";
import { Button } from "./../../../../components"; //Alias "@components";
import { classnames } from "./../../../../utils/component-utils"; // Alias "@utils/component-utils";
import {
  goToNextRound,
  setWhoLost,
} from "./../../../../models/tik-tak-boom/actions"; // Alias "@models/tik-tak-boom/actions";
import { players } from "./../../../../models/tik-tak-boom/props"; // Actions "@models/tik-tak-boom/props";
import { IState } from "./../../../../models/interfaces"; // Alias @models/interfaces
const styles = require("./styles.module.css");

interface IProps {
  players: {
    id: number;
    isActive: boolean;
    numOfBooms: number;
    name: string;
    playsNow: boolean | null;
  }[];
  goToNextRound: () => void;
  setWhoLost: (id: number) => void;
}

const _EndRound: FC<IProps> = ({
  players,
  goToNextRound,
  setWhoLost,
}): ReactElement => {
  const playerNow = players.find(player => player.playsNow);

  return (
    <div className={styles.endRoundContainer}>
      <div
        className={classnames(styles.whoLost, "extraLargeText", "main-color")}
      >
        Bomb boomed on {playerNow.name}'
        {playerNow.name.slice(-1).toLowerCase() === "s" ? "" : "s"} hands
      </div>
      <div className={styles.playersContainer}>
        {players.map(player => (
          <Button
            key={player.id}
            other={{ disabled: !player.isActive }}
            className={classnames(styles.player, "largeText", {
              "secondary-light": playerNow.id === player.id,
              "secondary-dark": playerNow.id !== player.id,
            })}
            onClick={() => player.isActive && setWhoLost(player.id)}
          >
            {player.name}
          </Button>
        ))}
      </div>
      <Button
        onClick={() => goToNextRound()}
        className={classnames(
          styles.proceedButton,
          "primary",
          "extraLargeText"
        )}
      >
        PROCEED
      </Button>
    </div>
  );
};

const EndRound = connect(
  createStructuredSelector<
    IState,
    {
      players: IProps["players"];
    },
    {
      players: IProps["players"];
    }
  >({
    players,
  }),
  { goToNextRound, setWhoLost }
)(_EndRound);

export { EndRound };
