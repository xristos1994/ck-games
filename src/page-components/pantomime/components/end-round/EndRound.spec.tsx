import React, { MouseEventHandler, ReactChild } from 'react';
import renderer from 'react-test-renderer';
import { _EndRound as EndRound } from './EndRound';

const Button = (
  { children, onClick, className='' } :
  { children: ReactChild | ReactChild[]; onClick: MouseEventHandler<HTMLButtonElement>; className: string }
) => {
  return (
    <button onClick={onClick} className={className || undefined}>
      {children}
    </button>
  );
};

jest.mock('@components/', () => ({
  __esModule: true,
  Button: Button
}));

const ScoreBoard = () => {
  return (
    <div>
      ScoreBoard
    </div>
  );
};

jest.mock('./../score-board', () => ({
  __esModule: true,
  ScoreBoard: ScoreBoard
}));

const teams = [
  { id: 1, score: 1, name: 'Team 1', playsNow: true, movieFound: true },
  { id: 2, score: 2, name: 'Team 2', playsNow: false, movieFound: false },
  { id: 3, score: 1, name: 'Team 3', playsNow: false, movieFound: false }
];
const goToNextRound = () => void 0;
const setIfMovieFound = (found: boolean) => void found;
const t = (label: string, param?: string[]) => label + (param ? param[0] : '');

describe('page-components/Pantomime/EndRound', () => {
  it('renders correctly', () => {
    let tree = renderer.create(
      <EndRound
        teams={teams}
        goToNextRound={goToNextRound}
        setIfMovieFound={setIfMovieFound}
        // eslint-disable-next-line @typescript-eslint/ban-ts-comment
        // @ts-ignore
        t={t}
      />
    ).toJSON();
    expect(tree).toMatchSnapshot();

    tree = renderer.create(
      <EndRound
        teams={teams.map(team => ({ ...team, movieFound: false }))}
        goToNextRound={goToNextRound}
        setIfMovieFound={setIfMovieFound}
        // eslint-disable-next-line @typescript-eslint/ban-ts-comment
        // @ts-ignore
        t={t}
      />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
