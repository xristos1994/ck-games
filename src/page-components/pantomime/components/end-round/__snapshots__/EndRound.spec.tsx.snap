// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`page-components/Pantomime/EndRound renders correctly 1`] = `
<div
  className="endRoundContainer"
>
  <div
    className="title"
  >
    Team, your turn completedTeam 1
  </div>
  <div>
    ScoreBoard
  </div>
  <div
    className="movieFoundButtons"
  >
    <button
      className="movieActionSelected"
      onClick={[Function]}
    >
      Movie Found
    </button>
    <button
      className="movieActionNotSelected"
      onClick={[Function]}
    >
      Movie Not Found
    </button>
  </div>
  <button
    className="proceedButton"
    onClick={[MockFunction]}
  >
    CONTINUE
  </button>
</div>
`;

exports[`page-components/Pantomime/EndRound renders correctly 2`] = `
<div
  className="endRoundContainer"
>
  <div
    className="title"
  >
    Team, your turn completedTeam 1
  </div>
  <div>
    ScoreBoard
  </div>
  <div
    className="movieFoundButtons"
  >
    <button
      className="movieActionNotSelected"
      onClick={[Function]}
    >
      Movie Found
    </button>
    <button
      className="movieActionSelected"
      onClick={[Function]}
    >
      Movie Not Found
    </button>
  </div>
  <button
    className="proceedButton"
    onClick={[MockFunction]}
  >
    CONTINUE
  </button>
</div>
`;
